Training/python/Phase2/Training.py

```diff
@@ -33,11 +33,16 @@
 
 class DeepTauModel(keras.Model):
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, loss=None, n_tau=None, **kwargs):
         super().__init__(*args, **kwargs)
         self.loss_tracker = keras.metrics.Mean(name="loss")
         self.pure_loss_tracker = keras.metrics.Mean(name="pure_loss")
         self.reg_loss_tracker = keras.metrics.Mean(name ="reg_loss")
+        if loss is None:
+            self.model_loss = TauLosses.tau_crossentropy_v2
+        else:
+            self.model_loss = getattr(TauLosses,loss)
+        self.n_tau = n_tau
```

Comment:
no need to pass n_tau to the constructor use `n_tau = tf.shape(x[0])[0]` inside train/test step

Reply:
This doesn't seem to work, at least not without running TF with eagerly. `tf.shape(x[0])[0]` returns `Tensor("strided_slice:0", shape=(), dtype=int32)`.
It gives me the error: `TypeError: x and y must have the same dtype, got tf.float32 != tf.int32`. (Though strangely, it works whether I make n_tau an actual int for float.)
And when trying to convert to int (or float): `TypeError: int() argument must be a string, a bytes-like object or a number, not 'Tensor'`

Reply:
Are you using `tf.cast` when you trying to convert int to float? E.g.
```python
n_tau = tf.cast(tf.shape(x[0])[0], x[0].dtype)
```

Reply:
No I wasn't, but it does work like this, thanks!